pgx-version-updater/src/main.rs

```diff
@@ -0,0 +1,407 @@
+use std::io::{BufRead, Write};
+use std::process::{Command, Stdio};
+use std::{env, path::PathBuf};
+use toml_edit::{value, Document, Table};
+use walkdir::{DirEntry, WalkDir};
+
+#[derive(Parser, Debug)]
+#[clap(author, version, about, long_about = None)]
+struct Args {
+    /// Additional Cargo.toml file to include for processing that can't be detected automatically
+    ///
+    /// Add multiple values using --include /path/foo/Cargo.toml --include /path/bar/Cargo.toml
+    #[clap(short, long)]
+    include_for_dep_updates: Vec<String>,
+
+    /// Exclude Cargo.toml files from [package] version updates
+    ///
+    /// Add multiple values using --exclude /path/foo/Cargo.toml --exclude /path/bar/Cargo.toml
+    #[clap(short, long)]
+    exclude_from_version_change: Vec<String>,
+
+    /// Version to be used in all updates
+    #[clap(short, long, required = true)]
+    update_version: String,
+
+    /// Do not make any changes to files
+    #[clap(short, long)]
+    dry_run: bool,
+
+    /// Output diff between existing file and changes to be made
+    #[clap(short, long)]
+    show_diff: bool,
+}
+
+// List of directories to ignore while Walkdir'ing. Add more here as necessary.
+const IGNORE_DIRS: &'static [&'static str] = &[".git", "target"];
+
+fn main() {
+    let args = Args::parse();
+    let current_dir = env::current_dir().expect("Could not get current directory!");
+
+    // Contains a set of package names (e.g. "pgx", "pgx-pg-sys") that wil be used
+    // to search for updatable dependencies later on
+    let mut updatable_package_names_set: HashSet<String> = HashSet::new();
+
+    // This will eventually contain every file we want to process
+    let mut files_to_process_set: HashSet<String> = HashSet::new();
+
+    // Keep track of which files to exclude from a "package version" change.
+    // For example, some Cargo.toml files do not need this updated:
+    //   [package]
+    //   version = "0.1.0"
+    //   ...
+    // Any such file is explicitly added via a command line argument.
+    // Note that any files included here are still eligible to be processed for
+    // *dependency* version updates.
+    let mut exclude_version_files_set: HashSet<String> = HashSet::new();
+    for file in args.exclude_from_version_change {
+        exclude_version_files_set.insert(fullpath(file).to_str().unwrap().to_string());
+    }
+
+    // Recursively walk down all directories to extract out any existing Cargo.toml files
+    for entry in WalkDir::new(&current_dir)
+        .into_iter()
+        .filter_entry(|e| is_not_excluded_dir(e))
+        .filter_map(|v| v.ok())
+    {
+        if is_cargo_toml_file(&entry) {
+            let filepath: String = fullpath(entry.path().display().to_string())
+                .display()
+                .to_string();
+
+            let mut output = format!(
+                "{} Cargo.toml file at {}",
+                "Discovered".bold().green(),
+                &filepath.cyan()
+            );
```

Comment:
It might make sense to batch this result set if we want to report it (or not report it without `--verbose`).

Reply:
Ah, yeah, good idea.